config/routes.rb

```diff
@@ -3,5 +3,7 @@
   resources :articles do
     resources :comments
   end
-  resources :users
+  resources :users, param: :_username
+  post '/auth/login', to: 'authentication#login'
+  get '/*a', to: 'application#not_found'
```

Comment:
Can you add more context to the purpose of this route?

Reply:
The route is for returning an error message "not found" if user does not exist. More context is in `not_found` action in the applications controller.